contracts/stellar-interchain-token-service/src/tests/migrate.rs

```diff
@@ -0,0 +1,589 @@
+    );
+    assert_eq!(
+        env.as_contract(&its_client.address, || {
+            storage::flow_out(&env, token_id, current_epoch)
+        }),
+        flow_out_amount
+    );
+}
+
+#[test]
+fn migrate_token_fails_with_invalid_token_id() {
+    let (env, its_client, _, _, _) = setup_env();
+    let upgrader_client = setup_upgrader(&env);
+    let owner = its_client.owner();
+    let non_existent_token_id = BytesN::random(&env);
+
+    let its_wasm_hash = env.deployer().upload_contract_wasm(ITS_WASM);
+    let token_manager_wasm_hash_v110 = env.deployer().upload_contract_wasm(TOKEN_MANAGER_WASM_V110);
+    let interchain_token_wasm_hash_v110 = env
+        .deployer()
+        .upload_contract_wasm(INTERCHAIN_TOKEN_WASM_V110);
+
+    assert_auth!(owner, its_client.upgrade(&its_wasm_hash));
+
+    let migration_data = CustomMigrationData {
+        new_token_manager_wasm_hash: token_manager_wasm_hash_v110,
+        new_interchain_token_wasm_hash: interchain_token_wasm_hash_v110,
+    };
+
+    env.as_contract(&its_client.address, || {
+        assert_ok!(InterchainTokenService::__migrate(&env, migration_data));
+    });
+
+    assert_contract_err!(
+        its_client
+            .mock_all_auths_allowing_non_root_auth()
+            .try_migrate_token(
+                &non_existent_token_id,
+                &upgrader_client.address,
+                &String::from_str(&env, VERSION_V110),
+            ),
+        ContractError::InvalidTokenId
+    );
+}
+
+#[test]
+fn migrate_succeeds_with_multiple_token_ids() {
+    let (env, its_client, _, _, _) = setup_env();
+    let upgrader_client = setup_upgrader(&env);
+    let owner = its_client.owner();
+    let (token_id_1, _) = setup_its_token(&env, &its_client, &owner, 100);
+    let token_id_2 = its_client.mock_all_auths().deploy_interchain_token(
+        &owner,
+        &BytesN::random(&env),
+        &TokenMetadata {
+            name: String::from_str(&env, "Token2"),
+            symbol: String::from_str(&env, "TOKEN2"),
+            decimal: 6,
+        },
+        &200,
+        &None,
+    );
+
+    let its_wasm_hash = env.deployer().upload_contract_wasm(ITS_WASM);
+    let token_manager_wasm_hash_v110 = env.deployer().upload_contract_wasm(TOKEN_MANAGER_WASM_V110);
+    let interchain_token_wasm_hash_v110 = env
+        .deployer()
+        .upload_contract_wasm(INTERCHAIN_TOKEN_WASM_V110);
+
+    let current_epoch = current_epoch(&env);
+
+    let token_manager_v100_1 = its_client.deployed_token_manager(&token_id_1);
+    let token_manager_v100_2 = its_client.deployed_token_manager(&token_id_2);
+
+    let interchain_token_v100_1 = its_client.interchain_token_address(&token_id_1);
+    let interchain_token_v100_2 = its_client.interchain_token_address(&token_id_2);
+
+    let token_config_1 = TokenIdConfigValue {
+        token_address: interchain_token_v100_1,
+        token_manager: token_manager_v100_1,
+        token_manager_type: TokenManagerType::LockUnlock,
+    };
```

Comment:
create a helper to set legacy token params and flow limit to reduce duplication in tests